src/trio/_subprocess.py

```diff
@@ -1067,86 +1081,157 @@ async def my_deliver_cancel(process):
-            shell: bool = False,
-            cwd: StrOrBytesPath | None = None,
-            env: Mapping[str, str] | None = None,
-            preexec_fn: Callable[[], object] | None = None,
-            restore_signals: bool = True,
-            start_new_session: bool = False,
-            pass_fds: Sequence[int] = (),
-        ) -> trio.Process: ...
-
-        @overload  # type: ignore[no-overload-impl]
-        async def run_process(
-            command: StrOrBytesPath,
-            *,
-            task_status: TaskStatus[Process] = trio.TASK_STATUS_IGNORED,
-            stdin: bytes | bytearray | memoryview | int | HasFileno | None = None,
-            capture_stdout: bool = False,
-            capture_stderr: bool = False,
-            check: bool = True,
-            deliver_cancel: Callable[[Process], Awaitable[object]] | None = None,
-            stdout: int | HasFileno | None = None,
-            stderr: int | HasFileno | None = None,
-            close_fds: bool = True,
-            shell: Literal[True],
-            cwd: StrOrBytesPath | None = None,
-            env: Mapping[str, str] | None = None,
-            preexec_fn: Callable[[], object] | None = None,
-            restore_signals: bool = True,
-            start_new_session: bool = False,
-            pass_fds: Sequence[int] = (),
-        ) -> subprocess.CompletedProcess[bytes]: ...
-
-        @overload
-        async def run_process(
-            command: Sequence[StrOrBytesPath],
-            *,
-            task_status: TaskStatus[Process] = trio.TASK_STATUS_IGNORED,
-            stdin: bytes | bytearray | memoryview | int | HasFileno | None = None,
-            capture_stdout: bool = False,
-            capture_stderr: bool = False,
-            check: bool = True,
-            deliver_cancel: Callable[[Process], Awaitable[None]] | None = None,
-            stdout: int | HasFileno | None = None,
-            stderr: int | HasFileno | None = None,
-            close_fds: bool = True,
-            shell: bool = False,
-            cwd: StrOrBytesPath | None = None,
-            env: Mapping[str, str] | None = None,
-            preexec_fn: Callable[[], object] | None = None,
-            restore_signals: bool = True,
-            start_new_session: bool = False,
-            pass_fds: Sequence[int] = (),
-        ) -> subprocess.CompletedProcess[bytes]: ...
+
+        class UnixProcessArgs(GeneralProcessArgs, total=False):
+            preexec_fn: Callable[[], object] | None
+            restore_signals: bool
+            start_new_session: bool
+            pass_fds: Sequence[int]
+
+            # 3.9+
+            group: str | int | None
+            extra_groups: Iterable[str | int] | None
+            user: str | int | None
+            umask: int
+
+        class UnixProcessArgs3_10(UnixProcessArgs, total=False):
+            pipesize: int
+
+        class UnixProcessArgs3_11(UnixProcessArgs3_10, total=False):
+            process_group: int
```

Comment:
typeshed and the docs show this as a default to `None` but that doesn't make sense because the docs also say:
> If `process_group` is a non-negative integer, the `setpgid(0, value)` system call will be made in the child process prior to the execution of the subprocess.

So why not make it default to `-1` like `pipesize`?

Reply:
I don't see any reason to deviate from typeshed/upstream, and there's ~no runtime difference

https://github.com/gpshead/cpython/blob/d8ca5a11bc55e2a69cab4f8795d0a5aa6932a41b/Lib/subprocess.py#L895